.github/workflows/InitDev,Unit tests,Integration tests,Package.yml

```diff
@@ -48,14 +49,38 @@ jobs:
       working-directory: ./Ed-Fi-ODS-Implementation/
       shell: pwsh
       run: |
-            $patternName = 'refs/heads/' +  '${{ env.CURRENT_BRANCH }}'
-            Write-Host "Pattern Name is " $patternName -fore GREEN
-            $is_pull_request_branch = 'False'
-            $is_pull_request_branch = git ls-remote --heads origin ${{ env.CURRENT_BRANCH }} | Select-String -Pattern $patternName -SimpleMatch -Quiet
-            if ($is_pull_request_branch -eq $true) {
-              git fetch origin ${{ env.CURRENT_BRANCH }}
-              git checkout ${{ env.CURRENT_BRANCH }}
-            }
+           $odsBranch = '${{ env.REPOSITORY_DISPATCH_BRANCH }}'
+           Write-Host "OdsBranch: $odsBranch"
+           if($odsBranch -ne ''){
+              $patternName = "refs/heads/$odsBranch"
+              $does_corresponding_branch_exist = $false
+              $does_corresponding_branch_exist = git ls-remote --heads origin $odsBranch | Select-String -Pattern $patternName -SimpleMatch -Quiet
+              if ($does_corresponding_branch_exist -eq $true) {
+                Write-Host "Corresponding branch for $odsBranch exists in Implementation repo, so checking it out"
+                git fetch origin $odsBranch
+                git checkout $odsBranch
+              } else {
+                WriteHost "Corresponding branch for $odsBranch does not exist in Implementation repo, so not changing branch checked out"
+              }
+           } else {
+              Write-Host "ref_name: ${{env.REF_NAME}}"
+              $current_branch = ${{env.REF_NAME}}
+              if ($current_branch -like "*/merge"){
+                Write-Host "ref_name is PR, so using head_ref: ${{env.HEAD_REF}}"
+                $current_branch = ${{env.HEAD_REF}}
+              }
+              $patternName = "refs/heads/$current_branch"
+              Write-Host "Pattern Name is " $patternName -fore GREEN
+              $branch_exists = $false
+              $branch_exists = git ls-remote --heads origin $current_branch | Select-String -Pattern $patternName -SimpleMatch -Quiet
+              if ($branch_exists -eq $true) {
+                Write-Host "Current branch exists, so setting to $current_branch"
+                git fetch origin $current_branch
+                git checkout $current_branch
+              } else {
+                Write-Host "did not match on any results for changing ODS checkout branch"
+              }
+           }
```

Comment:
What do you think about moving most of this script into a dedicated PowerShell file? This way it could be analyzed with https://github.com/PowerShell/PSScriptAnalyzer when we eventually introduce it (surprise!). That also makes it easier to test the script locally without running in GitHub Actions.

I like to put such scripts into a `eng` directory at the root of the repo. That said, I realize there's an existing "logistics" root and it would be fine to put files there instead, in keeping with the repo's current organization.

This could be deferred to another ticket if needed. And I am making a suggestion, not a demand 😁 .

Reply:
I agree! Especially since it is something I was already thinking about too.

Reply:
Okay this has been done.